tests/integrationtest/r/expression/time.result

```diff
@@ -1064,3 +1064,314 @@ a	d
+addtime(null, col2)
+NULL
+NULL
+NULL
+NULL
+NULL
+NULL
+NULL
+select subtime(null, col2) from t order by col0, col1 asc;
+subtime(null, col2)
+NULL
+NULL
+NULL
+NULL
+NULL
+NULL
+NULL
+set sql_mode='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+drop table if exists t;
+create table t(col0 datetime, col1 time, col2 varchar(30));
+insert into t values('0-0-0', '1:1:1', '1:1:1');
+set sql_mode='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+select addtime(col0, col1) from t;
+Error 1292 (22007): Incorrect date value: '0000-00-00 00:00:00'
+select subtime(col0, col1) from t;
+Error 1292 (22007): Incorrect date value: '0000-00-00 00:00:00'
+select addtime(col0, col2) from t;
+Error 1292 (22007): Incorrect date value: '0000-00-00 00:00:00'
+select subtime(col0, col2) from t;
+Error 1292 (22007): Incorrect date value: '0000-00-00 00:00:00'
+truncate t;
+set sql_mode='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+insert into t values('2024-1-0', '1:1:1', '1:1:1');
+set sql_mode='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+select addtime(col0, col1) from t;
+Error 1292 (22007): Incorrect date value: '2024-01-00 00:00:00'
+select subtime(col0, col1) from t;
+Error 1292 (22007): Incorrect date value: '2024-01-00 00:00:00'
+select addtime(col0, col2) from t;
+Error 1292 (22007): Incorrect date value: '2024-01-00 00:00:00'
+select subtime(col0, col2) from t;
+Error 1292 (22007): Incorrect date value: '2024-01-00 00:00:00'
+truncate t;
+set sql_mode='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+insert into t values('2024-0-1', '1:1:1', '1:1:1');
+set sql_mode='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+select addtime(col0, col1) from t;
+Error 1292 (22007): Incorrect date value: '2024-00-01 00:00:00'
+select subtime(col0, col1) from t;
+Error 1292 (22007): Incorrect date value: '2024-00-01 00:00:00'
+select addtime(col0, col2) from t;
+Error 1292 (22007): Incorrect date value: '2024-00-01 00:00:00'
+select subtime(col0, col2) from t;
+Error 1292 (22007): Incorrect date value: '2024-00-01 00:00:00'
+drop table if exists t;
+create table t(col0 date, col1 time, col2 varchar(30));
+set sql_mode='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+insert into t values('0-0-0', '1:1:1', '1:1:1');
+set sql_mode='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';
+select addtime(col0, col1) from t;
```

Comment:
add tests that the sql mode does contains NO_ZERO_IN_DATE/NO_ZERO_DATE?

Reply:
> add tests that the sql mode does contains NO_ZERO_IN_DATE/NO_ZERO_DATE?

done